core/base_algo.py

```diff
@@ -371,26 +372,31 @@ def get_remaining_categories(self) -> Set:
         return ALL_CATEGORIES - categories_in_file
 
     def _update_results_csv(self, values_dict) -> None:
-        if self.args.category in set(self.experiment_results_manager.data.category):
-            mask = (self.experiment_results_manager.data.category != self.args.category)
+        key = [self.args.category, self.model_name]
+        if self.experiment_results_manager.data[['category', 'model_name']].isin(key).any(axis=1).any():
+            # Key match found, overwriting
+            mask = (self.experiment_results_manager.data[['category', 'model_name']] != key)
             self.experiment_results_manager.data = self.experiment_results_manager.data[mask]
         
         new_dict = values_dict.copy()
         new_dict['category'] = self.args.category
         new_dict['category_type'] = CATEGORY_TO_TYPE[self.args.category]
+        new_dict['model_name'] = self.model_name
         self.experiment_results_manager.data = \
             pd.concat(
                 [self.experiment_results_manager.data, pd.DataFrame(new_dict, index=[0])]
-            ).reset_index(drop=True).sort_values(['category_type', 'category'])
+            ).reset_index(drop=True).sort_values(['model_name', 'category_type', 'category'])
 
     def _get_categories_in_data_file(self) -> Set:
         """
-        Getting a list of the categories from experiment_results_manager.
-        
+        Getting a list of the categories from experiment_results_manager that correspond
+        to the `self.model_name` model.
+
         Return: Set[str]
         -------
         A set of strings denoting the categories stored in the data file.
         """
-        categories = set(self.experiment_results_manager.data["category"])
+        categories = set(
+            self.experiment_results_manager.data.loc[self.experiment_results_manager.data["model_name"] == self.model_name, "category"].unique())
```

Comment:
break this oneliner down ffs.
extract some variables if needed.

Reply:
Its not that long... The member name is just really long... `self.experiment_results_manager.data` repeats twice and explodes the line's length

Reply:
imagine:
`categories = set(df.loc[df['model_name'] == self.model_name, 'category'].unique())`